src/dsp_tools/commands/xmlupload/iiif_client.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+
+import requests
+
+from dsp_tools.commands.xmlupload.models.input_problems import AllIIIFUriProblems
+from dsp_tools.commands.xmlupload.models.input_problems import IIIFUriProblem
+from dsp_tools.utils.uri_util import is_iiif_uri
+
+
+@dataclass(frozen=True)
+class IIIFUriValidator:
+    """Client handling communication with external IIIF-servers to do a health check."""
+
+    uri_list: list[str]
+
+    def validate(self) -> AllIIIFUriProblems | None:
+        """Validate the URI and return a list of problems if any."""
+        iiif_uri_problems = [res for uri in self.uri_list if (res := self._validate_one_uri(uri)) is not None]
+        if iiif_uri_problems:
+            return AllIIIFUriProblems(problems=iiif_uri_problems)
+        return None
+
+    def _validate_one_uri(self, uri: str) -> IIIFUriProblem | None:
+        """Check if the IIIF-server is reachable. If not, it returns information for error message."""
+        regex_has_passed = is_iiif_uri(uri)
+        response = self._make_network_call(uri)
```

Comment:
question: if it doesn't match the regex, does it actually make sense to attempt the network call? and if so, if that succeeds, is it really still a problem?

Reply:
See explanation below.